### PR TITLE
(Wallet) Improve flow to manually add NFTs

### PR DESCRIPTION
**Pull request by simoarpe**
## Resolves https://github.com/brave/brave-browser/issues/31399

Add new plus icon inside NFT section that opens `AddAssetActivity`.
If a new NFT asset is added correctly, the `NftGridFragment` will update accordingly.

This PR includes minor code clean-ups.

https://github.com/brave/brave-core/assets/3308503/a8cda233-933b-4c38-aa41-5b21b05fbdb8

**Note:** As demonstrated in the recording above, the Nft grid is refreshed **only** when a new NFT is added. If the user navigates back without adding an NFT the layout is (correctly) not refreshed.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

* Open Wallet app
* Navigate to NFT section
* Tap on the plus icon
* Add a new NFT and tap Add
* Observe the new NFT showing up in the grid

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/EditVisibleAssetsBottomSheetDialogFragment.java [hunk 1]
```diff
@@ -40,7 +40,6 @@
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
 
 import org.chromium.base.Log;
-import org.chromium.brave_wallet.mojom.AssetRatioService;
```
Comment (simoarpe) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/EditVisibleAssetsBottomSheetDialogFragment.java:43]: All the changes in this class `EditVisibleAssetsBottomSheetDialogFragment` are just minor improvements suggested by the IDE.

**Review by Pavneet-Sing — APPROVED** [review]

File: android/java/res/layout/fragment_nft_grid.xml [hunk 2]
```diff
@@ -54,6 +54,17 @@
...
+                android:id="@+id/fragment_nft_grid_btn_add_nft"
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:layout_marginEnd="16dp"
+                app:srcCompat="@drawable/ic_plus"
+                app:tint="@color/wallet_text_color"
+                android:contentDescription="@null"
+                android:background="?android:attr/selectableItemBackground"
+                />
+
```
Comment (Pavneet-Sing) [inline android/java/res/layout/fragment_nft_grid.xml:67]: nit: it probably would look better if the gap between plus and items can be reduced. Probably @kleantzogu might have some suggestions to improve the design, maybe different icon than the default figma one etc.

Comment (kleantzogu) [reply android/java/res/layout/fragment_nft_grid.xml:67]: We did some updates on Wallet 2.0. 
[Figma Link (Android)](https://www.figma.com/file/muVcmcu4RFNu0LkXe9VRDl/Wallet-2.0?type=design&node-id=5516-387547&mode=design&t=9ifWGAXggRfwrpia-4)
[Figma Prototype](https://www.figma.com/proto/muVcmcu4RFNu0LkXe9VRDl/Wallet-2.0?type=design&node-id=5164-359379&viewport=1178%2C-2210%2C0.28&t=9ifWGAXggRfwrpia-0&scaling=scale-down&starting-point-node-id=5516%3A389297&show-proto-sidebar=1)

Also attached is recording.

https://github.com/brave/brave-core/assets/2451237/b31876bc-e446-4a90-8962-3220d7e9e50a

Comment (simoarpe) [reply android/java/res/layout/fragment_nft_grid.xml:67]: Thanks! Merging this as is for now and tweaking the UI in a follow-up PR.

**Pull request merged**